### ISSUE TITLE
az resource group deployment running long after deployment is finished

### ISSUE THREAD

**Issue by colemickens** [Azure/azure-cli#1060]
``` shell
+ az resource group deployment create --name mickens57fc9003 --resource-group mickens57fc9003 --template-file-path /home/cole/code/colemickens/acs-labs/acstgen/output/mickens57fc9003/azuredeploy.json --parameters-file-path /home/cole/code/colemickens/acs-labs/acstgen/output/mickens57fc9003/azuredeploy.parameters.json
```

The deployment finished minutes ago according to `resources.azure.com` yet I'm still waiting for azure-cli to realize it.

**Comment by yugangw-msft** [comment]
Thanks for reporting this. Could you run with `--debug` next time, this way we can see the trace for why the command was stuck.

**Comment by tjprescott** [comment]
It might be a situation like app gateway, where the resource is created and then it is in an updating kind of state where you can do stuff on the portal but the CLI will keep polling (because the update is part of the LRO).

**Comment by colemickens** [comment]
I was checking the Deployment object in resources.azure.com and it said "Succeeded". Additionally, this was one of a dozen identical deployments -- most seemed to linger for 30-45 seconds after I could see it was finished. I assumed this was CRP specifying the polling duration to be 60 seconds or something.

The one I observed last night seemed to be buggy. I've updated my deployment script to tee the debug output to a file so I'll have it next time it repros.
